node/src/accountant/scanners/mod.rs

```diff
@@ -377,65 +377,74 @@ impl PayableScanner {
 
     fn separate_existent_and_nonexistent_fingerprints<'a>(
         &'a self,
-        sent_payables: &'a [&'a PendingPayable],
+        sent_payables: &[&'a PendingPayable],
     ) -> (Vec<PendingPayableMetadata>, Vec<PendingPayableMetadata>) {
-        fn sew_metadata(
-            ((rowid_opt, hash), pending_payable): ((Option<u64>, H256), &PendingPayable),
-        ) -> PendingPayableMetadata {
-            PendingPayableMetadata::new(&pending_payable.recipient_wallet, hash, rowid_opt)
-        }
-
         let hashes = sent_payables
             .iter()
             .map(|pending_payable| pending_payable.hash)
             .collect::<Vec<H256>>();
-
-        let sent_payables_sorted_by_hashes = sent_payables
+        let sent_payables_hashmap = sent_payables
             .iter()
-            .sorted_by(|a, b| Ord::cmp(&a.hash, &b.hash))
-            .copied()
-            .collect::<Vec<&PendingPayable>>();
+            .map(|payable| (payable.hash, &payable.recipient_wallet))
+            .collect::<HashMap<H256, &Wallet>>();
 
-        let rowid_pairs_sorted_by_hashes = self
-            .pending_payable_dao
-            .fingerprints_rowids(&hashes)
-            .into_iter()
-            .sorted_by(|(_, hash_a), (_, hash_b)| Ord::cmp(&hash_a, &hash_b))
-            .collect::<Vec<(Option<u64>, H256)>>();
+        let transaction_hashes = self.pending_payable_dao.fingerprints_rowids(&hashes);
+        let mut rowid_pairs_with_hashes = transaction_hashes
+            .rowid_results
+            .iter()
+            .map(|(_rowid, hash)| *hash)
+            .collect::<HashSet<H256>>();
+        for hash in &transaction_hashes.no_rowid_results {
+            rowid_pairs_with_hashes.insert(*hash);
+        }
+        let sent_payables_hashes = hashes.iter().copied().collect::<HashSet<H256>>();
 
-        if !Self::is_symmetrical(
-            &sent_payables_sorted_by_hashes,
-            &rowid_pairs_sorted_by_hashes,
-        ) {
+        if !PayableScanner::is_symmetrical(sent_payables_hashes, rowid_pairs_with_hashes) {
             panic!(
                 "Inconsistency in two maps, they cannot be matched by hashes. Data set directly \
                 sent from BlockchainBridge: {:?}, set derived from the DB: {:?}",
-                sent_payables_sorted_by_hashes, rowid_pairs_sorted_by_hashes
+                sent_payables, transaction_hashes.rowid_results
             )
         }
 
-        rowid_pairs_sorted_by_hashes
+        let pending_payables_with_rowid = transaction_hashes
+            .rowid_results
+            .into_iter()
+            .map(|(rowid, hash)| {
+                PendingPayableMetadata::new(
+                    <&Wallet>::clone(
+                        sent_payables_hashmap
+                            .get(&hash)
```

Comment:
Instead of get() there is a method remove() to be used on a HashSet. It means you might not need to clone the wallet at all. Look into it please.